llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -16102,33 +16102,7 @@ AArch64TargetLowering::BuildSDIVPow2(SDNode *N, const APInt &Divisor,
       !(Divisor.isPowerOf2() || Divisor.isNegatedPowerOf2()))
     return SDValue();
 
-  SDLoc DL(N);
-  SDValue N0 = N->getOperand(0);
-  unsigned Lg2 = Divisor.countr_zero();
-  SDValue Zero = DAG.getConstant(0, DL, VT);
-  SDValue Pow2MinusOne = DAG.getConstant((1ULL << Lg2) - 1, DL, VT);
-
-  // Add (N0 < 0) ? Pow2 - 1 : 0;
-  SDValue CCVal;
-  SDValue Cmp = getAArch64Cmp(N0, Zero, ISD::SETLT, CCVal, DAG, DL);
```

Comment:
The AArch64 code was using target specific nodes not generic nodes. This changes needs to be reviewed by AArch64 developers.

Reply:
Ping for review from AArch64 developers.

Reply:
Ping for review from AArch64 developers.

Reply:
Sorry for the delay, I didn't see this. I think it sounds fine so long as it is usually expected to optimize to the same thing.